Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CXFSAMLCallbackHandler.java

```diff
@@ -112,7 +114,7 @@ public void handle(final Callback[] callbacks) throws IOException, UnsupportedCa
                     final SamlTokenCreator creator = new SamlTokenCreator();
                     final CallbackProperties properties = new CallbackMapProperties(addMessageProperties(
                         creator.createRequestContext(custAssertion, getResource(message), null), message));
-                    oSAMLCallback.setAssertionElement(builder.build(properties, StoreUtil.getGatewayAlias(message)));
+                    oSAMLCallback.setAssertionElement(builder.build(properties, StoreUtil.getGatewayCertificateAlias(exchangeName)));
```

Comment:
can we used the `ExchangeManager` instead of `StoreUtil` to get teh gatewayAlias?

Reply:
No, the functionality is not the same. We want to grab it from the Map you set up from the URL implementation, not directly from the exchangeInfo.xml file. The key difference here is that it will default to "gateway" if its not in the map, whereas the ExchangeManager would return null

Reply:
you can always add the default-gateway instead of returning the null.

Reply:
the reason i used-the storeUtil instead of the exchangeManager is to seperated the logic; but now that we consolidating it down to the exchangeName i think it should go into the exchangemanager